operations/mimir-mixin/alerts/ingest-storage.libsonnet

```diff
@@ -19,6 +19,100 @@
             message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s is failing to commit the last consumed offset.' % $._config,
           },
         },
+
+        {
+          alert: $.alertName('IngesterFailedToReadRecordsFromKafka'),
+          'for': '5m',
+
+          // Metric used by this alert is reported by Kafka client on read errors from connection to Kafka.
+          // We use node_id to only alert if problems to the same Kafka node are repeating.
+          // If problems are for different nodes (eg. during rollout), that is not a problem, and we don't need to trigger alert.
+          expr: |||
+            sum by(%(alert_aggregation_labels)s, %(per_instance_label)s, node_id) (rate(cortex_ingest_storage_reader_read_errors_total[1m]))
+            > 0
+          ||| % $._config,
+          labels: {
+            severity: 'critical',
+          },
+          annotations: {
+            message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s is failing to read records from Kafka.' % $._config,
+          },
+        },
+
+        {
+          alert: $.alertName('IngesterKafkaFetchErrorsRateTooHigh'),
+          'for': '15m',
+          // See https://github.com/grafana/mimir/blob/24591ae56cd7d6ef24a7cc1541a41405676773f4/vendor/github.com/twmb/franz-go/pkg/kgo/record_and_fetch.go#L332-L366 for errors that can be reported here.
+          expr: |||
+            sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (rate (cortex_ingest_storage_reader_fetch_errors_total[5m]))
+            /
+            sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (rate (cortex_ingest_storage_reader_fetches_total[5m]))
+            > 0.1
+          ||| % $._config,
+          labels: {
+            severity: 'critical',
+          },
+          annotations: {
+            message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s is receiving fetch errors when reading records from Kafka.' % $._config,
+          },
+        },
+
+        // This is an experiment. We compute derivatition (ie. rate of consumption lag change) over 5 minutes. If derivation is above 0, it means consumption lag is increasing, instead of decreasing.
+        {
+          alert: $.alertName('StartingIngesterKafkaReceiveDelayIncreasing'),
+          'for': '5m',
+          expr: |||
+            deriv(histogram_quantile(0.99, sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (rate(cortex_ingest_storage_reader_receive_delay_seconds{phase="starting"}[1m])))[5m:1m]) > 0
```

Comment:
Thinking loudly: I'm wondering if it's better to use avg or 99th percentile here. Maybe there's no difference. Since we expect to consume records in order, the average (measured over the last 1m) should be good as well. Main differences that comes to my mind:
- 99th take longer to reduce while catching up, but also takes longer to increase when lag is increasing
- average doesn't suffer imprecise measurement given by 99th percentile when classic histogram is used

I have no strong opinion. I leave my thoughts to you. Same applies to `MimirRunningIngesterReceiveDelayTooHigh`.

Reply:
> average doesn't suffer imprecise measurement given by 99th percentile when classic histogram is used

Note that we explicitly use native histogram in the query here.

Reply:
Switched to using average (but still querying native histogram series).

Reply:
> but still querying native histogram series

Had to switch to computing average from classic histograms instead due to https://github.com/monitoring-mixins/mixtool/issues/163